테스트 코드 세팅

Unagi-zoso opened the pull request:
### Description

테스트 작성을 앞두고 가이드 겸 견본 테스트를 작성했습니다.
BDD 스타일을 따르며 통합테스트는 지양하시고 모키스트 테스트를 진행합시다. (스케쥴러 빈 분리를 미약하게 해서..)

<!--
  간단하게 PR의 목적을 설명하세요.
  이 PR이 해결하려는 문제나 추가하려는 기능에 대해 요약해주세요.
-->

### Related Issues

<!--
  관련된 이슈 번호를 참고하세요.
  예: Fixes #123, Closes #456
-->
- Resolves #110 

### Changes Made

<!--
  이 PR에서 변경된 사항을 설명하세요.
  코드, 문서, 설정 등 변경된 내용을 상세히 기술합니다.
-->

1. H2 디비 의존성 추가
2. application-test.yml 및 브랜치 보안 규칙
3. Admin 기능 테스트 코드 추가


### Screenshots or Video

<!--
  변경된 사항이 UI에 영향을 미치는 경우, 변경 전후의 스크린샷이나 동영상을 첨부하세요.
-->

### Testing

<!--
  변경 사항을 테스트한 방법을 설명하세요.
  테스트한 환경 (OS, 브라우저, 장치 등)과 테스트 절차를 구체적으로 기술합니다.
-->

1. 테스트 코드 작성
2. Gradle 테스트 코드 실행

### Checklist

<!--
  PR 작성 시 다음 항목들을 확인하세요.
-->

- [x] 코드가 정상적으로 컴파일되는지 확인했습니다.
- [x] 모든 테스트가 성공적으로 통과했습니다.
- [x] 관련 문서를 업데이트했습니다.
- [x] PR이 관련 이슈를 정확히 참조하고 있습니다.
- [x] 코드 스타일 가이드라인을 준수했습니다.
- [x] 코드 리뷰어를 지정했습니다.

### Additional Notes

<!--
  리뷰어가 이해하는 데 도움이 될 추가적인 참고 사항이나 정보가 있다면 여기에 작성하세요.
-->

Unagi-zoso committed:
chore: H2 DB 의존성 추가 및 application-test 파일 작성

테스트 편의를 위해 H2 DB 와 ddl-auto 를 활성화합니다.

Unagi-zoso committed:
fix: unique 컬럼 오탈자 수정 gongsi_id -> gongsi_code

누락된것이 있어 반영한다.

Unagi-zoso committed:
test: AdminControllerTest 작성

Unagi-zoso committed:
test: TodayProcessedGongsiManagerTest 작성

BDD 스타일을 따릅니다.

Unagi-zoso committed:
test: TodayProcessedGongsiServiceTest 작성

BDD 스타일을 따릅니다.

Unagi-zoso committed:
test: TodayProcessedGongsiRepositoryTest 작성

BDD 스타일을 따릅니다.

Unagi-zoso committed:
chore: PR , Push 테스트 워크플로우 작성

PR 시 테스트가 성공하는지 확인한다.

bakgom123 commented on src/test/java/org/bob/siungongsi/service/TodayProcessedGongsiManagerTest.java:
```diff
@@ -0,0 +1,116 @@
+
+  @Test
+  @DisplayName("생성 시 DB에서 공시 목록을 로드한다")
+  void whenConstruct_thenLoadFromDBToSet() {
```
생성자 테스트(whenConstruct_thenLoadFromDBToSet)는 초기화 로직을 검증하는 내용이므로, 테스트 클래스 맨 위쪽으로 옮기면 좀 더 자연스러울 것 같습니다.

Unagi-zoso replied:
좋은 의견이네요 👍

Unagi-zoso committed:
chore: H2 DB 드라이버 의존성 버전 명시

버전을 명시하는 편이 유지보수하기 좋아보여 명시합니다.

Unagi-zoso committed:
test: 테스트 코드 위치 변경

private 메소드인 loadFromDBToSet 을 테스트하기 위해 생성자를 사용하는데 이게 생성자를 테스트하는 성향이 강하다는 피드백이 있어 테스트 상단에 배치했습니다.

Unagi-zoso committed:
Merge branch 'develop' into test/init-setting
